### PR TITLE
ch4/fbox: Fix memory allocation

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

Add missing parentheses to correctly calculate the size of buffer
needed.

## Expected Impact

Fix address sanitizer warning.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [ ] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by raffenet** [comment]
test:jenkins/ch4/ofi

**Comment by raffenet** [comment]
test:mpich/warnings

File: src/mpid/ch4/shm/posix/eager/fbox/fbox_init.h [hunk 1]
```diff
@@ -53,7 +53,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_eager_init(int rank, int size)
     MPIR_CHKPMEM_MALLOC(MPIDI_POSIX_eager_fbox_control_global.first_poll_local_ranks,
                         int16_t *,
                         sizeof(*MPIDI_POSIX_eager_fbox_control_global.first_poll_local_ranks) *
-                        MPIR_CVAR_CH4_POSIX_EAGER_FBOX_POLL_CACHE_SIZE + 1, mpi_errno,
+                        (MPIR_CVAR_CH4_POSIX_EAGER_FBOX_POLL_CACHE_SIZE + 1), mpi_errno,
                         "fist_poll_local_ranks", MPL_MEM_SHM);
```
Comment (gcongiu) [inline src/mpid/ch4/shm/posix/eager/fbox/fbox_init.h]: Shall also fix the tag (`fist_poll_local_ranks -> first_poll_local_ranks`)?

Comment (raffenet) [reply src/mpid/ch4/shm/posix/eager/fbox/fbox_init.h]: Good catch. Will fix.

**Comment by raffenet** [comment]
test:jenkins/ch4/ofi
test:mpich/warnings

**Pull request merged**